### PR TITLE
[ROCM] GPU profiling code refactor - seperarted rocm event tracer and…

### PR DESCRIPTION
**Pull request by zahiqbal**
rocm profiler code refactored, Separated tracing and event collecting into rocm tracer (rocm_tracer.cc/h) and event collector (rocm_collector.cc/h).

### CODE REVIEW

File: xla/backends/profiler/gpu/rocm_collector.cc [hunk 1]
```diff
@@ -0,0 +1,889 @@
...
+    DumpRocmTracerEvent(api_event, 0, 0, "API_EVENT_RB");
+  }
+
+  for (auto& activity_iter : activity_ops_events_map_) {
+    RocmTracerEvent& activity_event = activity_iter.second.front();
+    if(activity_iter.second.size() > 1) {
+      LOG(INFO) << "RB: activity has more than event for correlation_id=" << activity_iter.first;
+    }
+    DumpRocmTracerEvent(activity_event, 0, 0, "ACTIVITY_EVENT_RB");
+  }*/
```
Comment (i-chaochen) [inline xla/backends/profiler/gpu/rocm_collector.cc:871]: shall we remove it?

File: xla/backends/profiler/gpu/rocm_collector.cc [hunk 2]
```diff
@@ -0,0 +1,889 @@
+
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```
Comment (i-chaochen) [inline xla/backends/profiler/gpu/rocm_collector.cc:2]: 2024 The OpenXLA Authors

File: xla/backends/profiler/gpu/rocm_collector.cc [hunk 3]
```diff
@@ -0,0 +1,889 @@
...
+        xevent.AddStatValue(*plane->GetOrCreateStatMetadata(
+                                GetStatTypeStr(StatType::kCorrelationId)),
+                            event.correlation_id);
+      }
+      if (!event.roctx_range.empty()) {
+        xevent.AddStatValue(*plane->GetOrCreateStatMetadata(
+                                GetStatTypeStr(StatType::kNVTXRange)),
+                            *plane->GetOrCreateStatMetadata(event.roctx_range));
+      }
+      // if (event.context_id != CuptiTracerEvent::kInvalidContextId) {
```
Comment (i-chaochen) [inline xla/backends/profiler/gpu/rocm_collector.cc:281]: shall we remove this?

File: xla/backends/profiler/gpu/rocm_collector.cc (hunk 1 above)
Comment (zahiqbal) [reply xla/backends/profiler/gpu/rocm_collector.cc:871]: sure, i will remove it

File: xla/backends/profiler/gpu/rocm_collector.cc (hunk 2 above)
Comment (zahiqbal) [reply xla/backends/profiler/gpu/rocm_collector.cc:2]: Fixed

**Comment by i-chaochen** [comment]
please check @xla-rotation Thanks in advance!

**Review by tdanyluk — APPROVED** [review]
It looks like these all only affect rocm, so I trust @i-chaochen's expertise and approve this after a quick check.

**Comment by i-chaochen** [comment]
Thanks @tdanyluk ! Yes, it's in our rocm side for JAX profiling.

**Pull request closed without merging**